p2719.md

```diff
@@ -809,11 +848,75 @@ For example, a special ADL lookup could be done based solely on the dynamic type
+inside `constexpr`. We apply the same rationale to `operator new[]`, `operator delete`, and
+`operator delete[]` in order to fully support type aware allocators where existing global allocators
+are supported.
+
+## Design choice: Stripping qualifiers from the type
+
+There is a question as to whether the type-identity should include qualifiers. In previous drafts
+this was not addressed, and retaining the qualifiers was implied. After consideration we have
+realised that this would be surprising, is inconsistent with how the type being allocated is
+presented to the developer throughout the process of allocation and deallocation, and results in the
+type-identity trivially diverging between such allocation and deallocation e.g.
+
+```cpp
+const T *t = new T;
+...
+delete t;
+```
+
+## Design choice: Safety restriction on mismatched declaration scopes
+
+This restriction was not discussed in Wroclaw, but we believe is a useful hazard restriction given
+the lack of any mechanism to otherwise force co-definition of new and delete operators. In [expr.new
+]{.unnumbered .sref} we would like to add the follow paragraph to force the selected deletion
+operator
+
+::: add
+> [31]{.pnum} If a single matching deallocation is found and either the allocation function or the
+> deallocation function has a type-identity parameter, then both functions shall belong to the same
+> scope ([basic.scope.scope]{.sref .unnumbered}), otherwise the program is ill formed.
+:::
+
+## Design choice: Permit sized delete when performing exception cleanup delete
+
+[@P3492R0] Discusses the problem of placement delete being unable to call sized delete. As a new
+syntax and function signature there is no risk of conflict with existing code. We could resolve
+[@P3492R0] by permitting a sized delete to be called during cleanup of placement new. The wording
+currently lacks this, and we lack direct implementation experience, but introducing support today
+might reduce problems in the future.
+
+The wording changes would likely be to [expr.new]{.sref .unnumbered}, something along the lines of
+
+> [29]{.pnum} A declaration of a placement deallocation function matches the declaration of a
+> placement allocation function if it has the same number of parameters and, after parameter
+> transformations ([dcl.fct]{.unnumbered .sref}), all parameter types except [the first]{.rm}[the
+> size and object]{.add}[, and optionally a parameter of type `std::size_t` immediately
+> following the address parameter]{.add} are identical.
+
+An alternative would be to mandate all currently "optional" parameters become mandatory for type
+aware allocation functions, however we feel that that considerably increases mental load of the
+feature, and more over that mandates sized delete, even though there are cases where it is not
+possible for the author or compiler to know the size of the object ahead of time.
```

Comment:
```suggestion

```

This opens up a significant design space. I think it's too late to try addressing this in C++26.